### DISCUSSION TITLE
Library for reading Menota XML?

### DISCUSSION THREAD

**Discussion by tarrinw** [network-digital-old-norse-studies/network-digital-old-norse-studies.github.io#25] [General]
I'm wondering if anyone has had much success reading Menota TEI/XML files in a programming library (especially Python, Perl, PHP or Javascript)?

I've had some success in PHP (cf. MenotaG), but this has always required resolving the entities before loading the file in PHP's library (using libxml2 as the backend). I've just tried using Python's lxml, but it seems to struggle with the standard Menota XML format.

It would make life easier if the library did the entity resolution itself.

I feel like Menota has worked on the principle that if it works in Oxygen, then it's fine. But open source / free applications and libraries don't seem to have much luck in working with these files, which limits how useful these editions are for digital analysis.

**Comment by tarrinw** [comment]
I should add that I've had some success with using the same libraries that Oxygen uses (especially Jing and Trang) to make my own applications do things that Oxygen otherwise only seems to be able to do, but I can't work out which ones listed in Oxygen resolve the other issues such as entity resolution (converting entity references into their corresponding characters).

**Comment by kraus-s** [comment]
Sorry, just seeing this now. I have written a menota parser for Python that will do entity resolution and was planning to release it along with my dissertation. I'd be happy to share it with you. It returns the parsed document as an object of a custom class. I'll get to it as soon as I can.

**Comment by kraus-s** [comment]
I have added you to the repository, Tarrin. If anyone else is interested, please let me know. I'll make the repository public at some point, but I want to get the code cleaned up just a bit more before I do that.
